### PR TITLE
feat(pagination): add pagination on list component

### PR DESCRIPTION
**Pull request by Peiren**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/List/Toolbar/Pagination/Pagination.component.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import css from './Pagination.scss';
+/**
+ * @param {object} props react props
+ * @example
+ <Pagination {...props}></Pagination>
+ */
+
+function Pagination({ itemsLength, sizeOptions, pageSize, activePage, onChange }) {
+	sizeOptions = sizeOptions || [5, 10, 20, 50];
+	pageSize = pageSize || sizeOptions[0];
```
Comment (yyanwang) [inline src/List/Toolbar/Pagination/Pagination.component.js]: I prefer to put these default props in Pagination.defaultProps.

File: src/List/Toolbar/Toolbar.component.js [hunk 2]
```diff
@@ -72,6 +74,7 @@ function Toolbar(props) {
 				{...sortProps}
 			/>
 			<Filter key="3" {...filterProps} />
+			<Pagination key="4" {...paginationProps} />
```
Comment (yyanwang) [inline src/List/Toolbar/Toolbar.component.js]: I prefer to make this Pagination component optional - if related props are not provided, then we don't add Pagination component in toolbar

File: src/List/Toolbar/Pagination/Pagination.component.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+			return;
+		}
+		onChange(from * pageSize, pageSize);
+	};
+
+	onChange(activePage * pageSize, pageSize);
+
+	return (
+		<Nav>
+			<label className="navbar-text" htmlFor="tc-list-toolbar-sort-by">Show:</label>
```
Comment (yyanwang) [inline src/List/Toolbar/Pagination/Pagination.component.js]: wrong id , I guess.

**Comment by jmfrancois** [comment]
you will have an issue because of the fix on the List story #111

File: src/List/Toolbar/Pagination/Pagination.component.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+			from = total - 1;
+			break;
+		}
+		default:
+			return;
+		}
+		onChange(from * pageSize, pageSize);
+	};
+
+	onChange(activePage * pageSize, pageSize);
```
Comment (talend-glorieux) [inline src/List/Toolbar/Pagination/Pagination.component.js]: Why do you need to call onChange here?

File: src/List/Toolbar/Pagination/Pagination.component.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+				))}
+			</NavDropdown>
+			<ButtonGroup className="navbar-btn">
+				<Button bsStyle="link" onClick={() => navTo(FIRST)} disabled={activePage === 0}>
+					<Icon name="fa fa-backward" />
+				</Button>
+				<Button bsStyle={classNames('link', css['tc-pagination-ctrl-prev'], 'tc-pagination-ctrl-prev')} onClick={() => navTo(PREV)} disabled={activePage === 0}>
+					<Icon name="fa fa-play" />
+				</Button>
+				<span className="btn btn-link">{activePage + 1}/{total}</span>
```
Comment (talend-glorieux) [inline src/List/Toolbar/Pagination/Pagination.component.js]: Please externalize computations into an external function. It makes it easier to handle edge cases.

File: src/List/Toolbar/Pagination/Pagination.component.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+				<Button bsStyle="link" onClick={() => navTo(NEXT)} disabled={activePage === (total - 1)}>
+					<Icon name="fa fa-play" />
+				</Button>
+				<Button bsStyle="link" onClick={() => navTo(LAST)} disabled={activePage === (total - 1)}>
```
Comment (talend-glorieux) [inline src/List/Toolbar/Pagination/Pagination.component.js]: Is there a way to make the FIRST and LAST button only display when there are something le 5/10 or more pages available. I don't really see its relevance when there are only 2/3 pages.

File: src/List/Toolbar/Pagination/Pagination.component.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * @param {object} props react props
+ * @example
+ <Pagination {...props}></Pagination>
+ */
+
+function Pagination({ itemsLength, sizeOptions, pageSize, activePage, onChange }) {
+	sizeOptions = sizeOptions || [5, 10, 20, 50];
+	pageSize = pageSize || sizeOptions[0];
+	const total = (itemsLength / pageSize) + (itemsLength % pageSize === 0 ? 0 : 1);
```
Comment (talend-glorieux) [inline src/List/Toolbar/Pagination/Pagination.component.js]: I get the logic behind the total calculation but I think its readability could be improved.

File: src/List/Toolbar/Pagination/Pagination.component.js (hunk 4 above)
Comment (Peiren) [inline src/List/Toolbar/Pagination/Pagination.component.js]: like put it inside componentDidMount(), make the initial call to retrieve data.
Ok, maybe we can do that out of pagination, in a container component.

File: src/List/Toolbar/Pagination/Pagination.component.js (hunk 7 above)
Comment (Peiren) [inline src/List/Toolbar/Pagination/Pagination.component.js]: now, change to Math.ceil(itemsLength / pageSize)

**Comment by yyanwang** [comment]
Seems my comments have been addressed..thanks.
@talend-glorieux Hi Geoffory, would you please also check?

**Pull request merged**